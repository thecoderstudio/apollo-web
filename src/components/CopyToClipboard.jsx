import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Button from './buttons/Button';
import copyImg from '../images/copy.svg'

const propTypes = {
  text: PropTypes.string.isRequired
}

const Wrapper = styled.div`
  grid-template-columns: [textfield] 1fr [copy] 50px;
  display: grid;
  height: 50px;
`;

const TextField = styled.div`
  grid-column: textfield;
  
  border: none;
  border-radius: 5px;
  color: white;
  font-family: 'B612', sans-serif;
  font-size: 1rem;
  font-weight: 600;
  cursor: pointer;
  text-align:left;
  padding: 15px;
  background-color: ${props => props.theme.lightBlack};
  text-overflow: ellipsis; 
  white-space: nowrap;
  overflow: hidden;
  
`;

const StyledButton = styled(Button)`
  grid-column: copy;
  height: 100%;
`;

const Icon = styled.img`
  text-align: center;
  display: block;
  width: 60%;
  margin-left: 20%;
  height: 60%;
`;

class CopyToClipboard extends React.PureComponent {
  constructor(props) {
    super(props);
    this.copyToClipboard = this.copyToClipboard.bind(this);
  }

  copyToClipboard() {
    navigator.clipboard.writeText(this.props.text);
  }

  render() {
    return(
      <Wrapper>
        <TextField>{this.props.text}</TextField>
        <StyledButton id='copyToClipboardButton' onClick={this.copyToClipboard}><Icon src={copyImg} /></StyledButton>
      </Wrapper>
    );
  };
}

CopyToClipboard.propTypes = propTypes;

export default CopyToClipboard;