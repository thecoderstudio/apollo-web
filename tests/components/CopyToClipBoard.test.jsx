import React from 'react';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import CopyToClipboard from "../../src/components/CopyToClipboard";

const mockStore = configureStore([]);

Object.assign(navigator, {
  clipboard: {
    writeText: () => {},
  },
});

function getComponent(store, props) {
  return renderer.create(
    <Provider store={store}>
      <CopyToClipboard {...props} />
    </Provider>
  );
}

describe('copy to clipboard', () => {
  let store = mockStore({});

  it("renders correctly", () => {
    const tree = getComponent(store, { text: 'test'}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('onClick copies to clipboard', () => {
  const spy = spyOn(navigator.clipboard, 'writeText');
    const component= getComponent(store, { text: 'test'});
    const instance = component.root;

    instance.findByProps({ id: 'copyToClipboardButton' }).props.onClick();
    expect(spy).toBeCalledWith('test');
  });
});