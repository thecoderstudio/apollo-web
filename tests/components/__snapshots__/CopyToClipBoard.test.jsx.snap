// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`copy to clipboard renders correctly 1`] = `
.c2 {
  max-height: 50px;
  padding: 12px;
  border: none;
  border-radius: 5px;
  color: white;
  font-family: 'B612',sans-serif;
  font-size: 1rem;
  font-weight: 700;
  cursor: pointer;
}

.c2:hover {
  opacity: 0.90;
}

.c2:focus {
  outline: none;
}

.c2:active {
  opacity: 0.80;
}

.c2:disabled:hover {
  opacity: 1;
  cursor: auto;
}

.c0 {
  grid-template-columns: [textField] 1fr [copy] 50px;
  display: grid;
  height: 50px;
}

.c1 {
  grid-column: textField;
  border: none;
  border-radius: 5px;
  color: white;
  cursor: text;
  padding: 15px;
  overflow: auto;
  white-space: nowrap;
  overflow-y: hidden;
  overflow-x: scroll;
}

.c1::-webkit-scrollbar {
  background: transparent;
}

.c3 {
  grid-column: copy;
  height: 100%;
}

.c4 {
  text-align: center;
  display: block;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    test
  </div>
  <button
    className="c2 c3"
    id="copyToClipboardButton"
    onClick={[Function]}
  >
    <i
      className="c4 fas fa-copy"
    />
  </button>
</div>
`;
